src/stores/pacbioLibraries.js

```diff
@@ -90,11 +116,22 @@ export const usePacbioLibrariesStore = defineStore('pacbioLibraries', {
       return Object.values(state.libraryState.libraries)
         .filter((library) => library.tube)
         .map((library) => {
-          const { id, request, tag, tube, ...attributes } = library
+          const { id, request, tag_id, tag, tube, ...attributes } = library
+          const tagId = tag_id ?? tag
+          // Get the tag group ID from the library's tag ID or the tag's group ID.
+          const tagGroupId = tagId
```

Comment:
The `libraryTags` in `libraryState` represents the tags associated with all existing libraries. Previously, the `librariesArray` getter, which returns a reformatted library object, was looking for tags only in `libraryTags`. However, with the new edit feature, we can now change the tag of a library to one that doesn't exist in any of the libraries.

Therefore, librariesArray needs to search for the tag first in libraryTags. If not found, it should then look for it in tagState->tags. It's important  that tagState->tags will only get populated if a fetchPacbioTagSets is called before, which may not happen in all the places where it's called. Hence, a search in both places is required to ensure that librariesArray returns the correct tag associated with all libraries."

Hope it make sense. Any other solution, might require a more detailed refactoring of store I suppose.

Reply:
OK I think I understand. In that case is it worth moving the TagSet fetch to the librariesIndex page so that they are always fully populated  in tagState? That way we should only have to check tagState.

Reply:
Yes. 
But, the only concern is, if we make it dependent solely on tagState, wouldn't it necessitate calling fetchPacbioTagSets before using the librariesArray getter?